cornice/service.py

```diff
@@ -153,12 +156,19 @@ class Service(object):
     list_arguments = ('validators', 'filters', 'cors_headers', 'cors_origins')
 
     def __repr__(self):
-        return u'<Service %s at %s>' % (self.name, self.path)
+        return u'<Service %s at %s>' % (
+            self.name, self.pyramid_route or self.path)
 
-    def __init__(self, name, path, description=None, cors_policy=None, depth=1,
-                 **kw):
+    def __init__(self, name, path=None, description=None, cors_policy=None,
+                 depth=1, pyramid_route=None, **kw):
         self.name = name
         self.path = path
+        self.pyramid_route = pyramid_route
+
+        if not self.path and not self.pyramid_route:
+            raise TypeError('You need to pass either path or '
+                            'pyramid_route arg')
```

Comment:
nit: Maybe `ValueError` instead?

Reply:
When argument is missing from call, python throws `TypeError`, so I followed that since proper arguments are missing. I can change to `ValueError` if you prefer that.

Reply:
good point! don't change